system/react/src/components/Banner.tsx

```diff
@@ -0,0 +1,126 @@
+import { Close, Product } from '@carbon/icons-react';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+
+import { Button } from './Button';
+
+export const classySelector = '.banner';
+
+export const baseStyles = css`
+  padding: var(--spacing-l4);
+  display: grid;
+  grid-template-columns: min-content auto min-content;
+  grid-template-areas: 'icon title close' 'icon content content' 'icon actions actions';
+
+  .banner-icon {
+    grid-area: icon;
+    padding-right: var(--spacing-l4);
+  }
+
+  .banner-title {
+    grid-area: title;
+  }
+
+  .banner-close {
+    grid-area: close;
+    cursor: pointer;
+  }
+
+  .banner-content {
+    grid-area: content;
+    padding: var(--spacing-l2) 0;
+
+    &:last-child {
+      padding: var(--spacing-l2) 0 0;
+    }
+  }
+
+  .banner-actions {
+    grid-area: actions;
+    display: flex;
+  }
+
+  button:not(:last-child) {
+    margin-right: var(--spacing-l4);
+  }
+
+  &[data-success] {
+    background-color: var(--success-surface);
+  }
+
+  &[data-info] {
+    background-color: var(--info-surface);
+  }
+
+  &[data-warning] {
+    background-color: var(--warning-surface);
+  }
+
+  &[data-neutral] {
+    background-color: var(--grey-100);
+  }
+`;
+
+export interface Props {
+  title: string;
+  description: string;
+  agree?: string;
+  dismiss?: string;
+  'data-single'?: boolean;
+  'data-dismissed'?: boolean;
+  'data-basic'?: boolean;
+  'data-success'?: boolean;
+  'data-info'?: boolean;
+  'data-warning'?: boolean;
+  'data-neutral'?: boolean;
+}
+
+const Wrapper = styled.div`
+  ${baseStyles}
+`;
+
+export function Banner({
+  title,
+  description,
```

Comment:
We shouldn't have any compound components like this unless absolutely necessary.

```suggestion
export const Banner = styled.div`
  ${baseStyles}
`;
```

Reply:
Not sure if I understood your review correctly. Can you tell me if the change was what you meant?

Reply:
Will resolve this thread and deal with it in the other one as this is outdated.